### ISSUE TITLE
Configuration

### ISSUE THREAD

**Issue by itabakh** [StefanoChiodino/niceprofit#1]
Would be great to have a configuration file at the top level of the project (wallet). Also having json files for each algorithm would help adjust the settings (threads...). A couple of things that would be a great addition.

**Comment by StefanoChiodino** [comment]
Yes, agreed. On the long run I was hoping to have it completely manage cpuminer so specifying the wallet address is a must. For now it just runs the benchmark and decide what's the most profitable algorithm! I also haven't looked at how the configuration of cpuminer itself would interact, I never used it myself but I imagine you could specify the thread count there!

**Comment by itabakh** [comment]
https://github.com/tpruvot/cpuminer-multi/blob/linux/cpuminer-conf.json

With cpuminer, you can set thread count, cpu affinity and other parameters. These parameters are important to consider as they will effect the benchmarks achieved.

Also the windows version actually uses a number of different miners. Would be good to use XMRig and XMR-Stak as two other good options. All of these allow the option to be loaded with a configuration file so it should be manageable.

**Comment by StefanoChiodino** [comment]
Would you not want most of this configuration to be the same for all algorithm though? That would make the benchmark more fair. What parameters you think would be useful to set for a specific algorithm?

Using different miners would be good, in fact GPU mining would be useful as well but I guess that would come later on in this project. What Nicehash have on Windows is actually pretty good, that's what I'm aiming at but will not be easy to replicate.

**Comment by itabakh** [comment]
Some algorithms have an expectation for a certain scratchpad size of L3 cache to be available (for example cryptonight, https://monero.stackexchange.com/questions/1110/where-can-i-find-a-description-of-the-cryptonight-hash-algorithm). Thus the number of threads should actually be controlled by amount of L3 cache available. For cryptonight more threads don't necessarily mean better performance.

I think starting with the same miner that can take different configuration files based on the algorithms. Then multiple cpu miners that can be set for each algorithm (with custom configuration files). Then generalize to gpu mining might make for a good pacing of features.
